### ISSUE TITLE
Problem with ESP32-Core v2.x.x

### ISSUE THREAD

**Issue by guidol70** [djbottrill/ESP32-Z80-Emulator#1]
Your Emulator is working fine when using ESP32-Core v1.0.6 - Thanks :)
But when using a newer ESP32-Core v2.x.x the SD-Commands like sdfiles
got problems :(

sdfiles does show the content of the directory (/downloads) but then the ESP32 does disconnect/restart.

Did you ever test a newer ESP32-Core? Or do you got an idea what to change?

While compiling the follwoing warnings/errors are shown:
In function 'bool SDprintDir(fs::FS&)':
.ino:504:1: warning: no return statement in function returning non-void [-Wreturn-type]

I also created a feature cudown version of this emulator to get it to work with a WeMOS D1R32 (normal ESP32) and a Deneyap Mini (ESP32-S2)

The ESP32-S2 does need ESP32-Core 2.x.x because it isnt supported in ESP32-Core v1.0.6

Many thanks for informations & ideas in advance :)

**Comment by djbottrill** [comment]
Hi,

I’ve just downloaded the V2 core so I’ll have a look and get back to you. I’ve also got a few improvements I need to add.

David

**Comment by djbottrill** [comment]
I've spent a day trying to get core 2.02 working on my TTGO T1 and T2 boards with no success I can't even get the SD card to be recognised. I've changed to a LOLIN32 board and everything seems to work OK except SD card access is very slow. 

I've now reverted to the 2.01 core and the SD card now works fine, no problem with SDfiles and speed is back to V1.06 performance.

V2 seems to be broken and in particular V2.02 seems to be very bad.

**Comment by guidol70** [comment]
Thanks for checking it up!
But for me v2.0.1/2.0.0 doesnt work better than v2.0.2/2.0.3rc1 :(
The WeMOS D1 R32 (ESP32) is a little bit picky about the SDCard, but does work fine when compiled with ESP32-Core v1.0.6

The Deneyap Mini (ESP32S2) do also work with a better fullsize-SDCard (SLC-Cell from Toshiba 8GB), but after showing sdfiles the Deneyap Mini does disconnect with every ESP32-v2.x.x-Core :(

I did think about changing sd.h to sdfat-Library, aber that does give me other errors (as far as I did read, ESP32-Support may be halted in sdfat in future :-( )

The normal "DIR" from SDCard does work fine with the ESP32S2, so I dont know whats the difference to the SD-commands.

Hopefully espressif do fix their banana-V2.x.x ESP32 in the future :(

Attached my "fork" e.g. the Cutdown-/Jr-Version for the WEMOS ;)
[ESP32-Z80-Jr.zip](https://github.com/djbottrill/ESP32-Z80-Emulator/files/8491803/ESP32-Z80-Jr.zip)
